storage/mysql/tree_storage.go

```diff
@@ -170,6 +169,7 @@ type treeTX struct {
 	treeType      trillian.TreeType
 	hashSizeBytes int
 	subtreeCache  *cache.SubtreeCache
+	dirty         []*storagepb.SubtreeProto
```

Comment:
I don't follow why we now have dirty / flush semantics in the storage layer.

Reply:
We did have it previously. `SubtreeCache` accumulated all writes, which were then flushed on `treeTX.Commit`.

I found it a bit cheaper to batch-write everything in one RPC rather than one RPC per shard. Also, after some long debugging of MySQL storage I found out that stuff can't be put to a MySQL transaction object concurrently, so this means I either need to lock `t.mu` and write (that's again a lot of lock contention, plus notice that it's shared with reads as well), or do a faster lock `t.mu` + append-to-slice (and only write once in the end).